Add new-style api reference docs for 1.5

pwittrock opened the pull request:
**Note: This will be hard to review the source by hand because everything checked in is generated from other files.  I can commit the source for the generated files in a separate PR.**

Changes here: https://deploy-preview-1818--kubernetes-io-vnext-staging.netlify.com/docs/reference/
See the "New Beta Docs" links in the Table of Contents

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1818)
<!-- Reviewable:end -->

pwittrock commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1818)
<!-- Reviewable:end -->

devin-donnelly commented on docs/api-reference/1_5/documents/_overview.md:
```diff
@@ -0,0 +1,92 @@
+# <strong>API OVERVIEW</strong>
+
+Welcome to the Kubernetes API.  You can use our API to access Kubernetes API endpoints to read and write Kubernetes
```
"You can use the Kubernetes API to read and write Kubernetes resource objects via a Kubernetes API endpoint."

pwittrock replied:
Done

devin-donnelly commented on docs/api-reference/1_5/documents/_overview.md:
```diff
@@ -0,0 +1,92 @@
+
+## Resource Categories
+
+This is a highlevel overview of the basic types of resources provide by the Kubernetes API and their primary functions.
```
highlevel -> high-level

pwittrock replied:
Done

devin-donnelly commented on docs/api-reference/1_5/documents/_overview.md:
```diff
@@ -0,0 +1,92 @@
+
+This is a highlevel overview of the basic types of resources provide by the Kubernetes API and their primary functions.
+
+**Workloads** *Run your containers*: These are responsible for managing and running your containers on the cluster.
```
**Workloads** are objects you use to manage and run your containers on the cluster.

pwittrock replied:
Done

devin-donnelly commented on docs/api-reference/1_5/documents/_overview.md:
```diff
@@ -0,0 +1,92 @@
+
+**Workloads** *Run your containers*: These are responsible for managing and running your containers on the cluster.
+
+**Discovery & LB Resources** *Access your containers*: These are responsible for stitching your workloads together into an externally accessible Loadbalanced Service.
```
**Discovery and Load-Balancing** resources are objects you use to "stitch" your workloads together into an externally accessible, load-balanced Service.

pwittrock replied:
Done

devin-donnelly commented on docs/api-reference/1_5/documents/_overview.md:
```diff
@@ -0,0 +1,92 @@
+
+**Discovery & LB Resources** *Access your containers*: These are responsible for stitching your workloads together into an externally accessible Loadbalanced Service.
+
+**Config & Storage Resources** *Inject initialization data into your containers*: These are responsible for injecting data into your applications and persisting data externally to your container.
```
**Configuration and Storage** resources are objects you use to inject initialization data into your applications, and to persist data that is external to your container.

pwittrock replied:
Done

devin-donnelly commented on docs/api-reference/1_5/documents/_overview.md:
```diff
@@ -0,0 +1,92 @@
+
+**Config & Storage Resources** *Inject initialization data into your containers*: These are responsible for injecting data into your applications and persisting data externally to your container.
+
+**Cluster Resources** *Manage your cluster*: These are responsible for defining configuration of the cluster itself, and are generally only used by cluster operators.
```
**Cluster Resource** objects define how the cluster itself is configured; these are typically used only by cluster operators.

pwittrock replied:
Done

devin-donnelly commented on docs/api-reference/1_5/documents/_overview.md:
```diff
@@ -0,0 +1,92 @@
+
+**Cluster Resources** *Manage your cluster*: These are responsible for defining configuration of the cluster itself, and are generally only used by cluster operators.
+
+**Metadata Resources** *Configure resource meta behavior*: These are responsible for configuring behavior of your other Resources within the Cluster.
```
**Metadata** resources are objects you use to configure the behavior of other resources within the cluster, such as...  (Fill in "such as" here)

pwittrock replied:
Done

devin-donnelly commented on docs/api-reference/1_5/documents/_cluster.md:
```diff
@@ -0,0 +1,5 @@
+# <strong>CLUSTER</strong>
+
+These are responsible for defining configuration of the cluster itself, and are generally only used by cluster operators.
```
"Cluster Resources are responsible for..."

devin-donnelly commented on docs/api-reference/1_5/documents/_config.md:
```diff
@@ -0,0 +1,11 @@
+# <strong>CONFIG & STORAGE</strong>
+
+These are responsible for injecting data into your applications and persisting data externally to your container.
```
"Configuration and Storage resources are responsible for..."

devin-donnelly commented on docs/api-reference/1_5/documents/_definitions.md:
```diff
@@ -0,0 +1,3 @@
+# <strong>DEFINITIONS</strong>
+
+These are Definitions for objects used by the above APIs
```
"This section contains definitions for objects used in the Kubernetes APIs."

devin-donnelly commented:
In general, replace the "These are" constructs in each resource category header as I've specified in the comments, and this is fine for me.

devin-donnelly commented:
Also, the "Getting Started" section of the kubectl command reference should probably add a paragraph about how "run" and "expose" are the most basic commands to get a workload running on your cluster and exposed to the outside world to serve requests. WDYT?

pwittrock commented:
> Also, the "Getting Started" section of the kubectl command reference should probably add a paragraph about how "run" and "expose" are the most basic commands to get a workload running on your cluster and exposed to the outside world to serve requests. WDYT?

Yeah, each of the sections could probably use a brief summary.  I can add that.

pwittrock commented:
Alright, I think that I addressed you comments.  I added some sections for the categories in the kubectl ref section.  I also added a section without the operations since Brian was asking for it.  It is the same as the api, but just missing the operations.

devin-donnelly commented:
Okay, LGTM.

pwittrock commented:
Woot.
